test/mlang/mexpr.mc

```diff
@@ -99,6 +116,35 @@ lang LetEval = Let + VarEval
     eval (cons (x, eval env t1) env) t2
 end
 
+lang RecLets = Var
+  syn Expr =
+  | TmRecLets ([(String, Expr)], Expr)
+end
+
+lang RecLetsEval = RecLets + VarEval + Fix + FixEval
+  sem eval (env : Env) =
+  | TmRecLets t ->
+    let bindings = t.0 in
+    let body = t.1 in
+    let foldli = lam f. lam init. lam seq.
+      (foldl (lam acc. lam x. (addi acc.0 1, f acc.0 acc.1 x)) (0, init) seq).1 in
+    utest foldli (lam i. lam acc. lam x. concat (concat acc (int2string i)) x) "" ["a", "b", "c"]
+      with "0a1b2c" in
```

Comment:
Is this meant as documentation about what `foldli` does, or was it left by accident?

Reply:
(also, does it make sense to put in the standard library?)

Reply:
I wrote it as a generic helper, so it felt sensible to test it as one. Not sure about putting it in the standard library. It doesn't tend to be a part of standard libraries for other functional languages, to my knowledge anyway. I don't really have any preference either way.

Reply:
I actually liked having the test there, since it helped me understand what it did (and it has negligible performance overhead, at least in theory). Maybe it's too early to introduce the `listExtras.mc` library ;)